test/libethereum/blockchain.cpp

```diff
@@ -83,20 +88,55 @@ void doBlockchainTests(json_spirit::mValue& _v, bool _fillin)
 		biGenesisBlock.verifyInternals(&rlpGenesisBlock.out());
 		o["genesisRLP"] = toHex(rlpGenesisBlock.out(), 2, HexPrefix::Add);
 
-		// construct blockchain
+		// construct true blockchain
 		TransientDirectory td;
-		BlockChain bc(rlpGenesisBlock.out(), td.path(), WithExisting::Kill);
+		BlockChain trueBc(rlpGenesisBlock.out(), td.path(), WithExisting::Kill);
 
 		if (_fillin)
 		{
 			BOOST_REQUIRE(o.count("blocks"));
 			mArray blArray;
+
+			blockSet genesis;
+			genesis.first = rlpGenesisBlock.out();
+			genesis.second = uncleList();
+			blockRLPs.push_back(genesis);
 			vector<BlockInfo> vBiBlocks;
 			vBiBlocks.push_back(biGenesisBlock);
+
+			size_t importBlockNumber;
 			for (auto const& bl: o["blocks"].get_array())
 			{
-				stateTemp = state;
 				mObject blObj = bl.get_obj();
+				BOOST_REQUIRE(blObj.count("blocknumber"));
+
+
+				//each time construct a new blockchain up to importBlockNumber (to generate next block header)
+				vBiBlocks.clear();
+				vBiBlocks.push_back(biGenesisBlock);
+
+				TransientDirectory td_stateDB, td_bc;
+				BlockChain bc(rlpGenesisBlock.out(), td_bc.path(), WithExisting::Kill);
+				State state(OverlayDB(State::openDB(td_stateDB.path())), BaseState::Empty, biGenesisBlock.coinbaseAddress);
+				importer.importState(o["pre"].get_obj(), state);
+				state.commit();
+
+				importBlockNumber = std::max((int)toInt(blObj["blocknumber"]), 1);
```

Comment:
This would mean that every blockchain filler file needs "blocknumber" as a parameter. 
Since "blocks" is an array in the json file, the index (+1) can be used as the block number, unless it is defined with the "blocknumber" parameter, in order to create block chain reorganization tests.